server/controllers/reports_proposed/data/debitor_group_report.js

```diff
@@ -0,0 +1,75 @@
+// reports_proposed/data/balance_sheet.js
+// Collects and aggregates data for the enterprise balance sheet
+var q       = require('q');
+var db      = require('../../../lib/db');
+var numeral = require('numeral');
+
+var formatDollar = '$0,0.00';
+var debitorGroupReportDate = new Date();
+
+exports.compile = function (options) {
+  'use strict';
+
+  var i18nDebitorGroupReport = options.language == 'fr' ? require('../lang/fr.json').DEBITOR_GROUP_REPORT : require('../lang/en.json').DEBITOR_GROUP_REPORT;
+  var deferred = q.defer(), context = {}, params = options.involveJournal === true ? [options.dg.account_id, options.dg.account_id] : [options.dg.account_id];
+
+  var sql2 =
+    'UNION ALL SELECT s.uuid as saleUUID, CONCAT(pr.abbr, s.reference) AS saleNumber, s.cost, DATE_FORMAT(s.invoice_date, \'%d-%m-%Y\') AS invoice_date, dg.name, ac.account_txt, SUM(pj.debit_equiv) AS debit, SUM(pj.credit_equiv) AS credit, pj.trans_id, ' +
+    'CONCAT(pt.first_name, \' \', pt.last_name) AS patientName FROM sale AS s JOIN posting_journal AS pj ON pj.inv_po_id = s.uuid JOIN account AS ac ON pj.account_id = ac.id JOIN debitor_group AS dg ' +
+    'ON dg.account_id = pj.account_id JOIN patient AS pt ON pt.debitor_uuid = s.debitor_uuid JOIN project AS pr ON pr.id = s.project_id WHERE pj.account_id =? GROUP BY s.uuid';
+  var addSql = options.involveJournal === true ? sql2 : '';
+  var sql = 
+    'SELECT s.uuid as saleUUID, CONCAT(pr.abbr, s.reference) AS saleNumber, s.cost, DATE_FORMAT(s.invoice_date, \'%d-%m-%Y\') AS invoice_date, dg.name, ac.account_txt, SUM(gl.debit_equiv) AS debit, SUM(gl.credit_equiv) AS credit, gl.trans_id, ' +
+    'CONCAT(pt.first_name, \' \', pt.last_name) AS patientName FROM sale AS s JOIN general_ledger AS gl ON gl.inv_po_id = s.uuid JOIN account AS ac ON gl.account_id = ac.id JOIN debitor_group AS dg ' +
+    'ON dg.account_id = gl.account_id JOIN patient AS pt ON pt.debitor_uuid = s.debitor_uuid JOIN project AS pr ON pr.id = s.project_id WHERE gl.account_id =? GROUP BY s.uuid ' + addSql;
+  var filterResults = function (results){
```

Comment:
It is better to use named function definitions (such as `function filterResults() { /* ... code .. */ }`) rather than anonymous functions (such as `var x = function () { /* ... code ... */ };`).  Not only is the first option more flexible, since it is hoisted to the top of the scope, any errors will be easier to debug (the error will print the function name it was thrown in).